builtin/json.mbt

```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 ///|
+#visibility(change_to="readonly", "Use smart constructor to make json")
```

Comment:
I've seen the term "smart constructor" used in other places as well, but it is not clear to me what that means.
I also searched the official MoonBit docs for this term, but it doesn't come up with a definition.
Could you please define what this means somewhere?

![smart-constructor-2025-04-17_08-02-17](https://github.com/user-attachments/assets/d4b592a3-ca72-4f72-8544-63869d98c74c)

Reply:
Smart constructor is a design pattern that creates the instances of enum types via helper functions... It's not documented because this concept comes from other programming languages, [possibly Haskell](https://wiki.haskell.org/index.php?title=Smart_constructors).